### PR TITLE
a small patch to update from linear_pipe to strict_pipe

### LINKED ISSUE #1536: Change `external_transition_{writer,reader}` to be a `Strict_pipe` (in `coda_lib.ml`)

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
This PR did a minor patch to update code in favor of Strict_pipe rather than Linear_pipe.

Checklist:

Closes #1536

### CODE REVIEW

File: src/lib/coda_lib/coda_lib.ml [hunk 1]
```diff
@@ -662,7 +662,8 @@ module Make (Inputs : Inputs_intf) = struct
   let create (config : Config.t) =
     trace_task "coda" (fun () ->
         let external_transitions_reader, external_transitions_writer =
-          Linear_pipe.create ()
+          Strict_pipe.create Synchronous
+          (* Linear_pipe.create () *)
```
Comment (bkase) [inline src/lib/coda_lib/coda_lib.ml]: remove

File: src/lib/coda_lib/coda_lib.ml [hunk 2]
```diff
@@ -765,9 +766,13 @@ module Make (Inputs : Inputs_intf) = struct
           Transition_router.run ~logger:config.log ~network:net
             ~time_controller:config.time_controller ~frontier_mvar ~ledger_db
             ~network_transition_reader:
+              (Strict_pipe.Reader.map external_transitions_reader
+                 ~f:(fun (tn, tm) -> (`Transition tn, `Time_received tm) ))
+              (*
               (Strict_pipe.Reader.of_linear_pipe
                  (Linear_pipe.map external_transitions_reader
                     ~f:(fun (tn, tm) -> (`Transition tn, `Time_received tm) )))
+              *)
```
Comment (bkase) [inline src/lib/coda_lib/coda_lib.ml]: remove

File: src/lib/coda_lib/coda_lib.ml [hunk 3]
```diff
@@ -792,7 +797,9 @@ module Make (Inputs : Inputs_intf) = struct
                  (External_transition.of_verified
                     (With_hash.data transition_with_hash)) )) ;
         don't_wait_for
-          (Linear_pipe.transfer_id (Net.states net) external_transitions_writer) ;
+          (Strict_pipe.transfer
+             (Strict_pipe.Reader.of_linear_pipe (Net.states net))
```
Comment (bkase) [inline src/lib/coda_lib/coda_lib.ml]: can you change the `Net`'s use of Linear pipe to Strict pipe as well?

File: src/lib/coda_lib/coda_lib.ml (hunk 1 above)
Comment (ghost-not-in-the-shell) [reply src/lib/coda_lib/coda_lib.ml]: Just to be sure, by `remove' you mean remove the comments here.

File: src/lib/coda_lib/coda_lib.ml (hunk 3 above)
Comment (ghost-not-in-the-shell) [reply src/lib/coda_lib/coda_lib.ml]: Sure, I can have a look on that.

File: src/lib/gossip_net/gossip_net.ml [hunk 4]
```diff
@@ -155,7 +155,9 @@ module Make (Message : Message_intf) :
         in
         let peer_events = Membership.changes membership in
         let broadcast_reader, broadcast_writer = Linear_pipe.create () in
-        let received_reader, received_writer = Linear_pipe.create () in
+        let received_reader, received_writer =
+          Strict_pipe.create Synchronous
```
Comment (bkase) [inline src/lib/gossip_net/gossip_net.ml]: this should be an `Overflow Drop_head`

File: src/lib/gossip_net/gossip_net.ml [hunk 5]
```diff
@@ -181,8 +183,9 @@ module Make (Message : Message_intf) :
                        mismatch due to forgery
                      *)
                     Linear_pipe.force_write_maybe_drop_head
```
Comment (bkase) [inline src/lib/gossip_net/gossip_net.ml]: then you can use `Strict_pipe.Writer.write` here, and wrap the whole thing in `to_linear_pipe`.

File: src/lib/pipe_lib/strict_pipe.mli [hunk 6]
```diff
@@ -76,6 +76,11 @@ module Reader : sig
 
     val two : 'a t -> 'a t * 'a t
   end
+
+  val partition_map3 :
```
Comment (bkase) [inline src/lib/pipe_lib/strict_pipe.mli]: Can you include a doccomment here (you can look at `val fold` for example)
that says that reads have to happen in the same order as writes or else deadlock

**Pull request merged**